### PR TITLE
Fixes Incorrect emit on Array[number]?.() #2347

### PR DESCRIPTION
**Pull request by alberrttt**
I messed up my git (sorry).  Here's the old PR  #2467

### CODE REVIEW

File: src/TSTransformer/nodes/transformOptionalChain.ts [hunk 1]
```diff
@@ -257,10 +259,12 @@ function transformOptionalChainInner(
 			if (item.callOptional) {
 				if (item.kind === OptionalChainItemKind.PropertyCall) {
 					baseExpression = luau.property(convertToIndexableExpression(baseExpression), item.name);
-				} else {
+				} else if (item.kind === OptionalChainItemKind.ElementCall) {
```
Comment (osyrisrblx) [inline src/TSTransformer/nodes/transformOptionalChain.ts:262]: Why did you change the else -> else if?

Comment (alberrttt) [reply src/TSTransformer/nodes/transformOptionalChain.ts:262]: i didnt know that it would be narrowed enough; ill remove it when i get back to my computer

**Comment by Dionysusnu** [comment]
We should add a regression unit test for this while we're at it
